e2e/test_helpers.go

```diff
@@ -266,6 +265,29 @@ func prepareAKSNode(ctx context.Context, s *Scenario) {
 		toolkit.LogDuration(ctx, totalElapse, 3*time.Minute, fmt.Sprintf("Node %s took %s to be created and %s to be ready", s.Runtime.VMSSName, toolkit.FormatDuration(creationElapse), toolkit.FormatDuration(readyElapse)))
 	}
 
+	if s.Config.PostProvisioningReboot != nil && *s.Config.PostProvisioningReboot {
+		vmssAboutToRebootTime := time.Now() // Record the start time
+		s.T.Log("Rebooting the VMSS VM to complete post-provisioning...")
+
+		restart, err := config.Azure.VMSSVM.BeginRestart(ctx, *s.Runtime.Cluster.Model.Properties.NodeResourceGroup, s.Runtime.VMSSName, "", nil)
+		require.NoError(s.T, err, "failed to start reboot vmss operation")
+		time.Sleep(60 * time.Second)
```

Comment:
and another hack, time.Sleep usually either leads to slow tests or to flaky tests

Reply:
Yeah, the time.Sleep didn't fix the issue. I can probably remove this whole if statement from the PR and try to get it working in a different one.

Reply:
Code removed.